cnudie/util.py

```diff
@@ -101,6 +108,40 @@ def diff_component_descriptors(
     )
 
 
+def label_diff(
+    left_labels: typing.List[cm.Label],
+    right_labels: typing.List[cm.Label],
+) -> LabelDiff:
+
+    label_diff = LabelDiff()
+
+    left_label_name_to_label = {l.name: l for l in left_labels}
+    right_label_name_to_label = {l.name: l for l in right_labels}
+
+    label_diff.labels_only_left = [
+        l for l in left_labels if l.name not in right_label_name_to_label.keys()
+    ]
+    label_diff.labels_only_right = [
+        l for l in right_labels if l.name not in left_label_name_to_label.keys()
+    ]
+
+    for left_group, right_group in _enumerate_group_pairs(
+        left_artifacts=left_labels,
+        right_artifacts=right_labels,
+    ):
+        if len(left_group) == 1 and len(right_group) == 1:
+            if left_group[0].value == right_group[0].value:
+                continue
+            else:
+                label_diff.label_pairs_changed.append(
+                    (left_group[0], right_group[0]),
+                )
+        else:
+            raise ValueError('only one label with the same name is allowed')
```

Comment:
would it maybe make sense to have the `enumerate_group_pairs`-function check for the maximum-allowed labels?

or even better, we should add such a check into the component-model (invalid component-descriptors should not even be allowed to be published in the first place).

for the special-case of unique names, it might be adequate to have a special codepath that will not return two-tuples of lists, but simple two-tuples of elements.